Scripts/RemediationScripts/Remediate-InvalidAADObjectRoleAssignments.ps1

```diff
@@ -213,7 +224,28 @@ function Remove-AzTSInvalidAADAccounts
         # Getting all permanent role assignments.
         $currentRoleAssignmentList = $currentRoleAssignmentList | Where-Object {![string]::IsNullOrWhiteSpace($_.ObjectId)};
         $currentRoleAssignmentList | select -Unique -Property 'ObjectId' | ForEach-Object { $distinctObjectIds += $_.ObjectId }
-    }
+
+        # Getting ASC reported deprecated account object ids.
+        $ascUri = "https://management.azure.com/subscriptions/$($subscriptionId)/providers/Microsoft.Security/assessments/00c6d40b-e990-6acf-d4f3-471e747a27c4?api-version=2020-01-01"
+        $method = [Microsoft.PowerShell.Commands.WebRequestMethod]::Get
+        $classicAssignments = [ClassicRoleAssignments]::new()
+        $headers = $classicAssignments.GetAuthHeader()
+        $ascDeprecated = [ASCDeprecatedAccounts]::new()
+        $response = $ascDeprecated.GetASCDeprecatedAccounts([string] $ascUri, [string] $method, [psobject] $headers)
+    
+        $ascDeprecatedAccountResponseAsString = $response.properties.additionalData.deprecatedAccountsObjectIdList.Trim("[").Trim("]")
+
+        if (![string]::IsNullOrWhiteSpace($ascDeprecatedAccountResponseAsString))
+        {
+            $ascDeprecatedAccountList = $ascDeprecatedAccountResponseAsString.Split(",").Trim("`"")
```

Comment:
Do you want to declare `ascDeprecatedAccountList` before L238 for cases where `ascDeprecatedAccountResponseAsString` is NULL/Empty?

Reply:
Declared the list 'ascDeprecatedAccountList' before.